File: app/src/main/java/com/mundi/databaseadmin/database/ListDatabases.java
package com.mundi.databaseadmin.database;

import android.util.Log;

import java.util.ArrayList;

public class ListDatabases {

    private static final String TAG = "ListDatabases";
    private BackendConnection backendConnection;
    public ListDatabases(String uri) {
        Log.d(TAG,"ListDatabases()");
        String cmd = "listdatabases";
        backendConnection = new BackendConnection(uri, cmd);
    }

    public ArrayList<String> getDatabases() {
        Log.d(TAG, "getDatabases()");
        return backendConnection.getArrayListString();
    }

    public boolean getResponse() {
        if (backendConnection.getResposeCode() == 200) {
            return true;
        }
        return false;
    }

}

File: app-kotlin/src/main/java/com/mundi/databaseadmin/database/ListDatabases.kt
package com.mundi.databaseadmin.database

import android.util.Log
import java.util.ArrayList

class ListDatabases(uri: String?) {
    private val backendConnection: BackendConnection
    val databases: ArrayList<String>
        get() {
            Log.d(TAG, "getDatabases()")
            return backendConnection.arrayListString
        }

    val response: Boolean
        get() = if (backendConnection.resposeCode == 200) {
            true
        } else false

    companion object {
        private const val TAG = "ListDatabases"
    }

    init {
        Log.d(TAG, "ListDatabases()")
        val cmd = "listdatabases"
        backendConnection = BackendConnection(uri, cmd)
    }
}